### DISCUSSION TITLE
How to get dap typescript to work with lazyvim

### DISCUSSION THREAD

**Discussion by ScopeSV** [LazyVim/LazyVim#1543] [Q&A]
Hi!

I cant for the life of me get dap adapter to work.
I've installed the extras plugin 
```lua
    { import = "lazyvim.plugins.extras.dap.core" },
```
When I try to run the debugger it just says:
```
Error on launch: Cannot launch program '{path}'; setting the 'outFiles' attribute might help.
```

In the documentation under "lang -> typescript" I found some dap configurations. 
I copied the configuration and pasted it into a new file under plugins.
I downloaded the vscode-js-debug and pointed the path to the local `dapDebugServer.js` but no luck.

Does anyone know the steps for getting this to work?

**Comment by folke** [comment]
dont copy stuff from extras. You need to additionally enable the typescript extra to get typescript debugging working.

**Comment by ScopeSV** [comment]
I also have the typescript extra package. That package works flawless in all my typescript projects.
Since the dap adapter didn't work, it was worth a shot to see if I was able to override the path to dapDebugServer to see if it helps, but as mentioned earlier, did not. :(

Could it be that it doesn't find the dapDebugServer? And if so, where should it be placed for the dap to locate it?
